### PR TITLE
Convert test annotation to test prefix

### PR DESCRIPTION
**Pull request by Bl00D4NGEL**
closes https://github.com/rectorphp/rector-phpunit/issues/45

### CODE REVIEW

**Comment by Bl00D4NGEL** [comment]
All checks have passed 🎉  @TomasVotruba  please review

File: tests/Rector/ClassMethod/ReplaceTestAnnotationWithPrefixedFunctionRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class SomeTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @test
+     */
+    public function testOnePlusOneShouldBeTwo()
+    {
+        $this->assertSame(2, 1+1);
+    }
```
Comment (TomasVotruba) [inline tests/Rector/ClassMethod/ReplaceTestAnnotationWithPrefixedFunctionRector/Fixture/fixture.php.inc:30]: Thanks for the rule :+1: 

It should remove the annotaiton too, as it's not needed anymore thanks to "test" prefix in method name:

```suggestion
    public function testOnePlusOneShouldBeTwo()
    {
        $this->assertSame(2, 1+1);
    }
```

Comment (Bl00D4NGEL) [reply tests/Rector/ClassMethod/ReplaceTestAnnotationWithPrefixedFunctionRector/Fixture/fixture.php.inc:30]: I wasn't sure if we should do it in the same rector or use a different one so I went with "simplest" first. I agree that we should probably have it in one rule :)

File: src/Rector/ClassMethod/ReplaceTestAnnotationWithPrefixedFunctionRector.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            return null;
+        }
+
+        if (! str_contains($docComment->getText(), '@test')) {
+            return null;
+        }
+
+        $node->name->name = 'test' . ucfirst($node->name->name);
+        $docComment = new Doc(
+            str_replace('@test', '', $docComment->getText()),
```
[the remark this answers is not in the document]
Comment (Bl00D4NGEL) [reply src/Rector/ClassMethod/ReplaceTestAnnotationWithPrefixedFunctionRector.php]: Good suggestion, thanks! :)

**Comment by samsonasik** [comment]
@TomasVotruba test error seems due to service `Symplify\PHPStanRules\Rules\IfNewTypeThenImplementInterfaceRule` removed

```
 1) Rector\PHPUnit\Tests\Rector\ClassMethod\AddDoesNotPerformAssertionToNonAssertingTestRector\AddDoesNotPerformAssertionToNonAssertingTestRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
_PHPStan_3e014c27f\Nette\DI\ServiceCreationException: Service (Symplify\PHPStanRules\Rules\IfNewTypeThenImplementInterfaceRule::__construct()): Class 'Symplify\PHPStanRules\Rules\IfNewTypeThenImplementInterfaceRule' not found.
```

**Comment by TomasVotruba** [comment]
I'm working on Symplify upgrade... give me 15 minutes

**Comment by TomasVotruba** [comment]
I've retriggered the CI... looks good :+1: 

@Bl00D4NGEL Thank you for the new rule :clap:

**Pull request merged**